src/Raven.Server/Documents/Subscriptions/SubscriptionStorage.cs

```diff
@@ -590,14 +600,35 @@ public bool TryGetRunningSubscriptionConnectionsState(long subscriptionId, out S
             if (subscriptionBlittable == null)
                 return null;
 
-            var subscriptionState = JsonDeserializationClient.SubscriptionState(subscriptionBlittable);
+            if (subscriptionBlittable.TryGet(nameof(SubscriptionState.SubscriptionId), out long id) == false)
+            {
+                if (_logger.IsOperationsEnabled)
+                    _logger.Info($"Could not figure out the Subscription Task ID for subscription named: '{subscriptionName}'.");
 
-            if (_subscriptions.TryGetValue(subscriptionState.SubscriptionId, out SubscriptionConnectionsState concurrentSubscription) == false)
+                return null;
+            }
+
+            if (_subscriptions.TryGetValue(id, out SubscriptionConnectionsState concurrentSubscription) == false)
                 return null;
 
             return concurrentSubscription;
         }
 
+        public class SubscriptionStateWithRaftIndex : SubscriptionState
+        {
+            // raft index used to create or update subscription task
+            public long RaftIndex { get; set; }
```

Comment:
I think a name like `LastModifiedIndex` would be more clear

Reply:
done